### ISSUE TITLE
Polygon overlap

### ISSUE THREAD

**Issue by eladvider** [manuelbieh/Geolib#77]
Are there any plans to add a function that checks if two polygons overlap?

**Comment by tony-gutierrez** [comment]
Yeah, being able to detect overlapping circles based on lat, lng, radius would be awesome.

**Comment by williammuff** [comment]
I might try to work on this, but essentially i think it would need to work with all type of shapes. For example: finding this overlap

<img width="519" alt="Untitled picture" src="https://user-images.githubusercontent.com/20742520/71768093-1438af80-2ee1-11ea-8870-9b64ab65642a.png">

**Comment by manuelbieh** [comment]
It's indeed hard to add such a function because as @williammuff already pointed out, it needs to work with all different kind of shapes. I had (and have) no immediate plans to add it but I'd be happy to merge a pull request if it's well tested.
